compiler/src/test/java/com/squareup/anvil/compiler/InterfaceMergerTest.kt

```diff
@@ -622,4 +623,221 @@ class InterfaceMergerTest(
+      assertThat(subcomponentInterface extends secondContributingInterface).isFalse()
+    }
+  }
+
+  @Test
+  fun `replaced interfaces contributed to multiple scopes must use the same scope`() {
+    assumeIrBackend()
+
+    compile(
+      """
+      package com.squareup.test
+      
+      import com.squareup.anvil.annotations.ContributesTo
+      $import
+      
+      @ContributesTo(Any::class)
+      @ContributesTo(Unit::class)
+      interface ContributingInterface
+      
+      @ContributesTo(Int::class, replaces = [ContributingInterface::class])
+      interface SecondContributingInterface
+      
+      $annotation(Any::class)
+      interface ComponentInterface
+      
+      $annotation(Unit::class)
+      interface SubcomponentInterface1
+      
+      $annotation(Int::class)
+      interface SubcomponentInterface2
+      """
+    ) {
+      assertThat(exitCode).isError()
+      assertThat(messages).contains(
+        "com.squareup.test.SecondContributingInterface with scope kotlin.Int wants to " +
+          "replace com.squareup.test.ContributingInterface, but the replaced class isn't " +
+          "contributed to the same scope."
+      )
+    }
+  }
+
+  @Test fun `interfaces contributed to multiple scopes can be excluded in one scope`() {
+    assumeIrBackend()
+
+    compile(
+      """
+      package com.squareup.test
+      
+      import com.squareup.anvil.annotations.ContributesTo
+      $import
+      
+      @ContributesTo(Any::class)
+      @ContributesTo(Unit::class)
+      interface ContributingInterface
+      
+      @ContributesTo(Any::class)
+      interface SecondContributingInterface
+      
+      $annotation(Any::class, exclude = [ContributingInterface::class])
+      interface ComponentInterface
+      
+      $annotation(Unit::class)
+      interface SubcomponentInterface
+      """
+    ) {
+      assertThat(componentInterface extends contributingInterface).isFalse()
+      assertThat(componentInterface extends secondContributingInterface).isTrue()
+      assertThat(subcomponentInterface extends contributingInterface).isTrue()
+      assertThat(subcomponentInterface extends secondContributingInterface).isFalse()
+    }
+  }
+
+  @Test fun `contributed interfaces in the old format are picked up`() {
+    val result = AnvilCompilation()
+      .useIR(USE_IR)
+      .enableAnvil(false)
+      .compile(
+        """
+        package com.squareup.test
+      
+        import com.squareup.anvil.annotations.ContributesTo
+        $import
+        
+        @ContributesTo(Any::class)
+        interface ContributingInterface  
+        """,
+        """
+        package anvil.hint.merge.com.squareup.test
+
+        import com.squareup.test.ContributingInterface
+        import kotlin.reflect.KClass
+        
+        public val com_squareup_test_ContributingInterface_reference: KClass<ContributingInterface> = ContributingInterface::class
+        
+        // Note that the number is missing after the scope. 
```

Comment:
I am also unfamiliar with what this part is referencing. Would we normally expect a number after the scope here?

Reply:
See the comment above. I should update the Anvil internals doc.

Reply:
@JoelWilcox I've updated the doc in the same PR. Does this make sense?